codegen/gateway.go

```diff
@@ -650,3 +749,16 @@ func (gateway *GatewaySpec) GenerateMain() error {
 	)
 	return err
 }
+
+// GenerateMiddlewareSchemas will generate the middleware schema files
+// for the gateway
+func (gateway *GatewaySpec) GenerateMiddlewareSchemas() error {
```

Comment:
I may not have enough context here, but I don't think zanzibar has to handle middleware schema. Here is what i think will happen:

1. Middleware developer creates a go package for a middleware(can put it any repo), and put the middleware options schema in a known repo for UI to locate the its `name`, `Go package path` and `schema`.

2. endpoint creators are shown a page for editing middleware `options` based on schema, verify it is a valid schema(possibly run Zanzibar also) and save it as a JSON file in the gateway repository.

Zanzibar only needs to generate code with  `name`, `Go package path` and `options` regardless where the option comes from.